net-libs/nativebiginteger/nativebiginteger-0.9.42.ebuild

```diff
@@ -0,0 +1,70 @@
+# Copyright 2018-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit java-pkg-2 toolchain-funcs
+
+DESCRIPTION='jbigi library used by net-vpn/i2p'
+HOMEPAGE='https://geti2p.net'
+SRC_URI="https://download.i2p2.de/releases/${PV}/i2psource_${PV}.tar.bz2"
+
+LICENSE='public-domain'
+SLOT='0'
+KEYWORDS='~amd64 ~x86'
+IUSE='test'
+
+DEPEND='
+	dev-libs/gmp:0=
+	virtual/jdk:1.8
+'
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/i2p-${PV}/core"
+
+PATCHES=(
+	"${FILESDIR}/${P}-asmfix.patch"
+)
+
+src_compile() {
+	local compile_lib
+	compile_lib() {
+		local name="${1}"
+		local file="${2}"
+		shift 2
+
+		"$(tc-getCC)" "${@}" ${CFLAGS} $(java-pkg_get-jni-cflags) \
+			${LDFLAGS} -shared -fPIC "-Wl,-soname,lib${name}.so" \
+			"${file}" -o "lib${name}.so"
+	}
+
+	cd "${S}/c/jbigi/jbigi" &&
```

Comment:
You'll need to split this and compile_lib.

Reply:
Why so? If either `cd` or `compile_lib` fails, `die` will be called.

Reply:
Even if it works like that, you can't figure which part failed exactly.